compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUInterfaces.td

```diff
@@ -60,6 +60,48 @@ def IREEGPU_MmaInterfaceAttr : AttrInterface<"MmaInterfaceAttr"> {
       /*methodName=*/"getABCVectorTypes",
       /*args=*/(ins)
     >,
+    InterfaceMethod<
```

Comment:
For every `InterfaceMethod` added in any new PR, I ask: why can't this simply be a global function declared in a `.h` instead? Two concerns here:
1. The C++ debate between member and non-member functions. It's been an evolution in the whole C++ culture (some googling found [this post](https://www.reddit.com/r/cpp/comments/262it3/comment/chn3vhw/) which summarizes a few links) and in the C++ standard library itself (e.g. `vector.begin()` vs `std::begin(vector)`)
2. In this instance it's also about going through generated code vs not. With `InterfaceMethod`, the indirection through generated code makes it harder to follow the code, both for the human reader, and even for tools like `clangd` which can't tell how many stack frames to walk to when you click "go to implementation".

In this instance, these were existing global functions, so why change that now?

Reply:
I think it's done, LMK if this is what you have in mind or if you think we can clean it up some more!